rust_book_ex_8_3/src/vector_ex.rs

```diff
@@ -0,0 +1,28 @@
+use std::collections::HashMap;
+
+pub fn get_mean(v: Vec<u32>) -> f32{
+    let sum: u32 = v.iter().sum();
+    return sum as f32 / v.len() as f32;
+}
+
+pub fn get_median(v: Vec<u32>) -> u32 {
+    let size = v.len();
+    return v[size / 2];
+}
+
+pub fn get_mode(v: Vec<u32>) -> u32 {
+    let mut number_counter: HashMap<u32, u32> = HashMap::new();
```

Comment:
If you don't want to write `HashMap` twice, you could also instantiate it as

```rust
let mut counter = HashMap::<u32, u32>::new();
```

(but that's not necessarily better here; just saying)